cpp/inc/bond/core/bond_version.h

```diff
@@ -16,12 +18,10 @@ namespace bond
     template <typename BufferT>
     class CompactBinaryReader;
 
-    static const uint16_t v1 = 0x0001;
-    static const uint16_t v2 = 0x0002;
+    BOND_STATIC_CONSTEXPR uint16_t v1 = 0x0001;
```

Comment:
I believe global `const` variables have [internal linkage](http://en.cppreference.com/w/cpp/language/storage_duration) implicitly (which is needed here, since these are _defined_ in a header). The `static` here makes it more explicit. The `const`/`constexpr` should be enough. I'll try to change and see if I hit any compiler issues.